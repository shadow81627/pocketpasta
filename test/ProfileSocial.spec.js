import component from '@/components/ProfileSocial';
import { shallowMount } from '@vue/test-utils';

const factory = () =>
  shallowMount(component, {
    propsData: {
      description:
        'Programmer and full stack web developer at XCOM Media, Griffith University graduate. Aspiring entrepreneur with a passion for startup weekends and hackathons.',
      keywords: 'html5, css3, jquery, bootstrap4',
      email: 'mailto:damien.robinson@pocketpasta.com',
      jobTitle: 'Full Stack Web Developer',
      brand: 'Pocket Pasta',
      name: 'Damien Robinson',
      givenName: 'Damien',
      familyName: 'Robinson',
      alumniOf: 'Griffith University',
      gender: 'male',
      nationality: 'Australian',
      url: 'https://pocketpasta.com',
      sameAs: [
        'https://github.com/shadow81627',
        'https://www.linkedin.com/in/damien-robinson-788925101/',
        'https://twitter.com/Shadow81627',
      ],
    },
  });

describe('ProfileSocial', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('head', () => {
    expect(component.head()).toBeTruthy();
  });
});
