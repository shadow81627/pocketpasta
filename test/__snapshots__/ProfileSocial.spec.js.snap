// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileSocial renders properly 1`] = `"<div class=\\"mb-3 profile\\"><div class=\\"col-md-12\\"><div class=\\"$col-md-6\\"><h2>Damien Robinson</h2> <key-value-stub label=\\"Job Title\\" value=\\"Full Stack Web Developer\\"></key-value-stub> <key-value-stub label=\\"About\\" value=\\"Programmer and full stack web developer at XCOM Media, Griffith University graduate. Aspiring entrepreneur with a passion for startup weekends and hackathons.\\"></key-value-stub> <keywords-stub label=\\"Skills\\" tags=\\"html5, css3, jquery, bootstrap4\\"></keywords-stub> <div class=\\"py-2 list-inline\\"><div class=\\"list-inline-item a2a_kit a2a_kit_size_32 a2a_default_style a2a_follow\\"><a data-a2a-follow=\\"Shadow81627\\" rel=\\"noopener\\" class=\\"a2a_button_twitter\\"></a> <a data-a2a-follow=\\"damien.robinson.5036\\" rel=\\"noopener\\" class=\\"a2a_button_facebook\\"></a> <a data-a2a-follow=\\"damien-robinson-788925101\\" rel=\\"noopener\\" class=\\"a2a_button_linkedin\\"></a> <a data-a2a-follow=\\"shadow81627\\" rel=\\"noopener\\" class=\\"a2a_button_youtube\\"></a> <a data-a2a-follow=\\"shadow81627\\" rel=\\"noopener\\" class=\\"a2a_button_github\\"></a></div></div></div></div></div>"`;
